### PR TITLE
Fixed content not restored on undo when multiple blocks and widgets were removed

### LINKED ISSUE #8870: Undo: paragraphs not restored when undoing image+text deletion
## 📝 Provide detailed reproduction steps (if any)

1. Set data where you have a paragraph with text, an image, and some other paragraphs with text.
2. Start removing data using backspace key from the end to the start.
3. Try undoing all deletions.

### ✔️ Expected result

Final content after undo is as it was at the beginning.

### ❌ Actual result

All but one paragraph after an image are missing.

## 📃 Other details

https://user-images.githubusercontent.com/1502228/105157521-58f3b080-5b0d-11eb-83d9-8f321b6cf68e.mp4

### PR DESCRIPTION
**Pull request by scofalik**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Fix (engine): Fixed content not restored on undo when multiple blocks and widgets were removed. Closes #8870.

### CODE REVIEW

**Comment by scofalik** [comment]
I know that there are no new tests but I modified an existing test and current result is better than old result.

File: packages/ckeditor5-engine/tests/model/operation/transform/merge.js [hunk 1]
```diff
@@ -78,7 +78,7 @@ describe( 'transform', () => {
 
 				john.undo();
 				syncClients();
-				expectClients( '<paragraph>Foo</paragraph><paragraph>Bar</paragraph><paragraph></paragraph>' );
+				expectClients( '<paragraph>Foo</paragraph><paragraph></paragraph><paragraph>Bar</paragraph>' );
```
Comment (scofalik) [inline packages/ckeditor5-engine/tests/model/operation/transform/merge.js:81]: The old expectation was incorrect (worse UX wise) than the new one.

**Comment by scofalik** [comment]
@ckeditor/qa-team Please please check :)

**Review by niegowski — COMMENTED** [review]
LGTM 👍  Waiting for confirmation from QA.

**Pull request merged**